src/main/java/org/commcare/util/CollectionUtils.java

```diff
@@ -28,4 +29,19 @@
         }
         return result;
     }
+
+    /**
+     * Checks if any element in {@code subList} is contained in {@code superList}
+     * @param superList
+     * @param subList
+     * @return true if {@code superList} contains any element of {@code subList}, false otherwise
+     */
+    public static boolean containsAny(ArrayList<String> superList, ArrayList<String> subList) {
+        for (String item : subList) {
+            if(superList.contains(item)){
```

Comment:
What are the expected upper bounds for the sizes of these lists? This is O(M*N) time complexity, right? Could that cause performance issues since `containsAny()` is called in a loop in `ShortestCycleAlgorithm.java`?

Reply:
@millerdev That's a good point. I think the size of these lists should not go beyond O(100) for most practical purposes. Also this code only gets called to single out the nodes constituting the cycle when we have already detected that there is a cycle that exists in the form. So this will end up getting called only in scenarios where forms have errors(cycles) making me think we should not worry about performance here too much.